x-pack/plugins/security_solution/public/management/pages/trusted_apps/store/middleware.ts

```diff
@@ -359,7 +379,27 @@ const fetchEditTrustedAppIfNeeded = async (
 
     // See if we can get the Trusted App record from the current list of Trusted Apps being displayed
     trustedAppForEdit = getListItems(currentState).find((ta) => ta.id === editTrustedAppId);
-    if (trustedAppForEdit) {
+
+    try {
+      // Retrieve Trusted App record via API if it was not in the list data.
+      // This would be the case when linking from another place or using an UUID for a Trusted App
+      // that is not currently displayed on the list view.
+      if (!trustedAppForEdit) {
+        dispatch({
+          type: 'trustedAppCreationEditItemStateChanged',
+          payload: {
+            type: 'LoadingResourceState',
+            // No easy way to get around this that I can see. `previousState` does not
+            // seem to allow everything that `editItem` state can hold, so not even sure if using
+            // type guards would work here
+            // @ts-ignore
+            previousState: editItemState(currentState)!,
```

Comment:
is `previousState` defined by us?  Could we expand it to expect `editItem` in this case?  Just trying to understand the reason for `ts-ignore` more clearly

Reply:
Its related to the conversation I (we) had with Bohdan about possibly changing the type for `previousState` to allow any of the `AsyncResourceState` types. Right now, `previousState` can hold a stale state, which is defined as:

https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/x-pack/plugins/security_solution/public/management/pages/trusted_apps/state/async_resource_state.ts#L75-L78

This is missing `LoadingResourceState` which can be a value assigned to the `editItem` property of state:

https://github.com/elastic/kibana/blob/2f97de897495df9007e08b148e1f1d1e48fcccdc/x-pack/plugins/security_solution/public/management/pages/trusted_apps/state/trusted_apps_list_page_state.ts#L58

Also - because `editItem` is defined as optional, `undefined` is also a possible value for it.

After the feedback we got from Bohdan, I think we'll probably make two changes:
- change this property `previousState` to just accept `AsyncResourceState` (instead of just a "stale" group), and
- make `previousState` optional

 But that change might have great amount of refactoring needed

Reply:
thanks, can you make a ticket for this if there isn't already?  Will 👍